design/httpproxy-include-route-duplicates-design.md

```diff
@@ -0,0 +1,301 @@
+In this API, there is no Status Condition set or error signalled to the user when there are duplicate route matches.
+The documentation lays out the expectations on conflict resolution between different route resources and within an HTTPRoute.
+We could take a similar documentation-only approach to route matching precedence, but it might leave some users with invalid configuration, confusion, etc.
+This would make it simpler to report status as writing complex header/query/path match conditions in status messages can be unwieldy.
+
+Note: this specification may change upstream in Gateway API, currently there are no conformance tests for it so as those are added we may see changes in this area.
+
+### Error condition messages without match conditions
+Instead of trying to enumerate exactly which set of match conditions caused the duplicate in the error message, we could use the json path syntax of the route to do so.
+For example we could have an error message such as `Match conditions for spec.routes[2] are a duplicate of spec.routes[0]` for errors within an HTTPProxy.
+Or: `Match conditions for spec.routes[2] are a duplicate of HTTPRoute foo/bar spec.routes[0]` for errors across HTTPProxies.
+
+This might lead to complications in accounting where routes came from which could get unwieldy to implement, but some combination of this and the currently proposed solution could be used.
+
+### Add a name field to routes
+This would make generating the status messages for errors easier, we could refer to a named route instead of trying to stringify the match conditions.
+However, if only used for this purpose, this might be heavyweight solution, since we would likely need to add uniqueness validation to some extent.
+
+One point in favor of this in a general API design perspective is that Envoy has a name field for routes that is mostly used for debugging in a similar vein as this addition would be.
+
+### Only update Status on HTTPProxy leaf that has an error
+This would reduce the ability for operators/administrators to follow up on issues in their site across included HTTPProxies.
+However this would reduce possible status update sprawl that may come from one bad actor/mistake/etc.
+
+## Security Considerations
+
+### Exposing match conditions from parent HTTPProxies down to child status
+We propose adding details of the full set of duplicate match conditions to be included in Status Conditions detailing why a route is invalid.
+This may be information not explicitly known/shared with the owner of the child HTTPProxy.
+Since a backend configured on such a route will receive a request with all of that information, I do not think it is a security concern to include that detail in most cases, but something to think about in case any organizations have an issue with this.
```

Comment:
I dont think it should be a problem, but adding a contour config flag to toggle this would be nice.